File: CX_Chat-main/backend/app/schemas/final_report.py
from pydantic import BaseModel


class FinalReportHero(BaseModel):
    report_title: str
    report_date_label: str | None = None
    company_name: str | None = None
    sector_name: str | None = None
    region: str | None = None
    overall_level: int | None = None
    overall_level_label: str | None = None
    overall_maturity_band: str
    hero_message: str | None = None
    strongest_axis: str | None = None
    strongest_axis_level: int | None = None
    strongest_axis_level_label: str | None = None
    priority_axis: str | None = None
    priority_axis_level: int | None = None
    priority_axis_level_label: str | None = None


class FinalReportSummary(BaseModel):
    overall_score_percent: float
    overall_maturity_band: str
    strongest_axis: str
    strongest_axis_score_percent: float
    priority_axis: str
    priority_axis_score_percent: float
    strengths_count: int
    pain_points_count: int
    assessed_capabilities_count: int = 0
    unassessed_capabilities_count: int = 0
    executive_summary_text: str | None = None
    priority_message_text: str | None = None


class FinalReportAxisItem(BaseModel):
    axis: str
    score_percent: float
    maturity_band: str
    axis_level: int | None = None
    axis_level_label: str | None = None


class FinalReportThemeItem(BaseModel):
    axis: str
    capability: str
    maturity_band: str
    rationale: str | None = None
    recommendation: str | None = None
    priority: str | None = None


class FinalReportCapabilityItem(BaseModel):
    capability_id: int | None = None
    maturity_level_number: int | None = None
    axis: str
    capability: str
    maturity_band: str
    assessment_status: str = "not_assessed"
    confidence: float | None = None
    rationale: str | None = None
    recommendation: str | None = None
    priority: str | None = None


class FinalReportBenchmarkItem(BaseModel):
    title: str
    url: str
    site_name: str | None = None
    published_at: str | None = None
    summary: str | None = None
    method_signal: str | None = None
    context_match: str | None = None


class FinalReportCompetitiveEvidenceLink(BaseModel):
    label: str
    url: str
    source_title: str | None = None


class FinalReportCompetitiveCompetitor(BaseModel):
    key: str
    company_name: str
    note: str | None = None
    stage_level: int
    stage_label: str
    logo_url: str | None = None
    is_you: bool = False
    evidence_links: list[FinalReportCompetitiveEvidenceLink] = []


class FinalReportCompetitiveStage(BaseModel):
    level: int
    label: str
    summary: str
    competitors: list[FinalReportCompetitiveCompetitor] = []


class FinalReportLeaderEvidenceLink(BaseModel):
    label: str
    url: str
    source_title: str | None = None
    mapped_capability: str | None = None
    why_relevant: str | None = None


class FinalReportLeadersSnapshotMetrics(BaseModel):
    candidates_considered: int = 0
    candidates_evaluated: int = 0
    web_search_calls: int = 0
    rerank_calls: int = 0
    mistral_calls: int = 0
    documents_retrieved: int = 0
    documents_validated: int = 0
    documents_rejected_indirect: int = 0
    capability_coverage_count: int = 0


class FinalReportLeaderItem(BaseModel):
    key: str
    company_name: str
    note: str | None = None
    leader_summary: str | None = None
    logo_url: str | None = None
    evidence_links: list[FinalReportLeaderEvidenceLink] = []


class FinalReportLeadersSnapshot(BaseModel):
    supported: bool
    status: str = "unavailable"
    sector: str
    respondent_company_name: str
    message: str | None = None
    metrics: FinalReportLeadersSnapshotMetrics | None = None
    leaders: list[FinalReportLeaderItem] = []


class FinalReportWorkingMissingItem(BaseModel):
    capability: str
    maturity_band: str
    summary: str | None = None
    evidence_snippet: str | None = None


class FinalReportWorkingMissingAxis(BaseModel):
    axis: str
    label: str
    score_percent: float
    maturity_band: str
    axis_level: int | None = None
    axis_level_label: str | None = None
    subtitle: str | None = None
    intro: str | None = None
    stat_note: str | None = None
    working: list[FinalReportWorkingMissingItem] = []
    missing: list[FinalReportWorkingMissingItem] = []


class FinalReportQuickWinItem(BaseModel):
    step: int
    timeline_label: str
    title: str
    owner: str
    today_text: str
    after_text: str


class FinalReportQuickWinsTimeline(BaseModel):
    section_title: str = "Your Quick Wins, In Order"
    items: list[FinalReportQuickWinItem] = []


class FinalReportResponse(BaseModel):
    assessment_id: int
    hero: FinalReportHero
    summary: FinalReportSummary
    axes: list[FinalReportAxisItem]
    strengths: list[FinalReportThemeItem]
    pain_points: list[FinalReportThemeItem]
    capabilities: list[FinalReportCapabilityItem]
    benchmarks: list[FinalReportBenchmarkItem]
    competitive_landscape: list[FinalReportCompetitiveStage] = []
    leaders_snapshot: FinalReportLeadersSnapshot | None = None
    quick_wins_timeline: FinalReportQuickWinsTimeline | None = None
    working_missing: list[FinalReportWorkingMissingAxis] = []

File: CX_Chat/frontend/src/types/final-report.ts
export type FinalReportHero = {
  report_title: string;
  report_date_label?: string | null;
  company_name?: string | null;
  sector_name?: string | null;
  region?: string | null;
  overall_level?: number | null;
  overall_level_label?: string | null;
  overall_maturity_band: string;
  hero_message?: string | null;
  strongest_axis?: string | null;
  strongest_axis_level?: number | null;
  strongest_axis_level_label?: string | null;
  priority_axis?: string | null;
  priority_axis_level?: number | null;
  priority_axis_level_label?: string | null;
};

export type FinalReportSummary = {
  overall_score_percent: number;
  overall_maturity_band: string;
  strongest_axis: string;
  strongest_axis_score_percent: number;
  priority_axis: string;
  priority_axis_score_percent: number;
  strengths_count: number;
  pain_points_count: number;
  assessed_capabilities_count?: number;
  unassessed_capabilities_count?: number;
  executive_summary_text?: string | null;
  priority_message_text?: string | null;
};

export type FinalReportAxisItem = {
  axis: string;
  score_percent: number;
  maturity_band: string;
  axis_level?: number | null;
  axis_level_label?: string | null;
};

export type FinalReportThemeItem = {
  axis: string;
  capability: string;
  maturity_band: string;
  rationale?: string | null;
  recommendation?: string | null;
  priority?: string | null;
};

export type FinalReportCapabilityItem = {
  capability_id?: number | null;
  maturity_level_number?: number | null;
  axis: string;
  capability: string;
  maturity_band: string;
  assessment_status?: string;
  confidence?: number | null;
  rationale?: string | null;
  recommendation?: string | null;
  priority?: string | null;
};

export type FinalReportBenchmarkItem = {
  title: string;
  url: string;
  site_name?: string | null;
  published_at?: string | null;
  summary?: string | null;
  method_signal?: string | null;
  context_match?: string | null;
};

export type FinalReportCompetitiveEvidenceLink = {
  label: string;
  url: string;
  source_title?: string | null;
};

export type FinalReportCompetitiveCompetitor = {
  key: string;
  company_name: string;
  note?: string | null;
  stage_level: number;
  stage_label: string;
  logo_url?: string | null;
  is_you?: boolean;
  evidence_links: FinalReportCompetitiveEvidenceLink[];
};

export type FinalReportCompetitiveStage = {
  level: number;
  label: string;
  summary: string;
  competitors: FinalReportCompetitiveCompetitor[];
};

export type FinalReportLeaderEvidenceLink = {
  label: string;
  url: string;
  source_title?: string | null;
  mapped_capability?: string | null;
  why_relevant?: string | null;
};

export type FinalReportLeadersSnapshotMetrics = {
  candidates_considered: number;
  candidates_evaluated: number;
  web_search_calls: number;
  rerank_calls: number;
  mistral_calls: number;
  documents_retrieved: number;
  documents_validated: number;
  documents_rejected_indirect: number;
  capability_coverage_count: number;
};

export type FinalReportLeaderItem = {
  key: string;
  company_name: string;
  note?: string | null;
  leader_summary?: string | null;
  logo_url?: string | null;
  evidence_links: FinalReportLeaderEvidenceLink[];
};

export type FinalReportLeadersSnapshot = {
  supported: boolean;
  status: string;
  sector: string;
  respondent_company_name: string;
  message?: string | null;
  metrics?: FinalReportLeadersSnapshotMetrics | null;
  leaders: FinalReportLeaderItem[];
};

export type FinalReportWorkingMissingItem = {
  capability: string;
  maturity_band: string;
  summary?: string | null;
  evidence_snippet?: string | null;
};

export type FinalReportWorkingMissingAxis = {
  axis: string;
  label: string;
  score_percent: number;
  maturity_band: string;
  axis_level?: number | null;
  axis_level_label?: string | null;
  subtitle?: string | null;
  intro?: string | null;
  stat_note?: string | null;
  working: FinalReportWorkingMissingItem[];
  missing: FinalReportWorkingMissingItem[];
};

export type FinalReportQuickWinItem = {
  step: number;
  timeline_label: string;
  title: string;
  owner: string;
  today_text: string;
  after_text: string;
};

export type FinalReportQuickWinsTimeline = {
  section_title: string;
  items: FinalReportQuickWinItem[];
};

export type FinalReport = {
  assessment_id: number;
  hero: FinalReportHero;
  summary: FinalReportSummary;
  axes: FinalReportAxisItem[];
  strengths: FinalReportThemeItem[];
  pain_points: FinalReportThemeItem[];
  capabilities: FinalReportCapabilityItem[];
  benchmarks: FinalReportBenchmarkItem[];
  competitive_landscape: FinalReportCompetitiveStage[];
  leaders_snapshot?: FinalReportLeadersSnapshot | null;
  quick_wins_timeline?: FinalReportQuickWinsTimeline | null;
  working_missing: FinalReportWorkingMissingAxis[];
};
